src/platform/baytrail/platform.c

```diff
@@ -239,6 +239,12 @@ int platform_init(struct sof *sof)
 #if defined CONFIG_CHERRYTRAIL
 	/* mask SSP 3 - 5 interrupts */
 	shim_write(SHIM_PIMRH, shim_read(SHIM_PIMRH) | 0x00000700);
+
+	/* Reset M/N SSP clock dividers */
+	shim_write(SHIM_SSP2_DIVL, 1);
+	shim_write(SHIM_SSP2_DIVL + 2, 0x80010000);
+	shim_write(SHIM_SSP2_DIVH, 0x80000001);
+	shim_write(SHIM_SSP2_DIVH + 2, 0x18000);
```

Comment:
My understanding is that we dont need the 16bit offset writes, we should be able to write to L then H (if the M/N is disabled), but I've never tried this before using 32bit ISA so it may not work. Thinking more, this could be also be fixed in the driver using a single 64 bit write directly per SSP port as part of the DSP reset/boot sequence ?.

Reply:
Man where is my brain, somehow I equated 32bits to 2 bytes and then started doing 2 writes to make up the difference. Clearly I forgot my coffee. I will fix this up on monday.

Reply:
The definition for those bits are

Divider_N: bit 19.0: default 0x1: Denominator value
Divider_M: bit 51:32: default 0x1: Nominator value
Divider Update: bit 61:61: default 0x0: Update divider
Divider_En: bit 62:62: default 0x0: Enable Divider
Divider_Bypass: bit 63:63: default 0x1: bypass divider.

We should make sure the values are reset to those defaults, so the value would be

0x8000 0001 0000 0001